src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -0,0 +1,310 @@
+    { id: ALLOY_IFRAME_ID, element: iframe }
+  ].forEach(({ id, element }) => {
+    const { style = {}, params = {} } = webParameters[id];
+
+    assign(element.style, style);
+
+    const {
+      parentElement = "body",
+      insertionMethod = "appendChild",
+      enabled = true
+    } = params;
+
+    const parent = document.querySelector(parentElement);
+    if (enabled && parent && typeof parent[insertionMethod] === "function") {
+      parent[insertionMethod](element);
+    }
+  });
+};
+
+export const buildStyleFromMobileParameters = mobileParameters => {
+  const {
+    verticalAlign,
+    width,
+    horizontalAlign,
+    backdropColor,
+    height,
+    cornerRadius,
+    horizontalInset,
+    verticalInset,
+    uiTakeover = false
+  } = mobileParameters;
+
+  const style = {
+    width: width ? `${width}%` : "100%",
+    backgroundColor: backdropColor || "rgba(0, 0, 0, 0.5)",
+    borderRadius: cornerRadius ? `${cornerRadius}px` : "0px",
+    border: "none",
+    position: uiTakeover ? "fixed" : "relative",
+    overflow: "hidden"
+  };
+  if (horizontalAlign === "left") {
+    style.left = horizontalInset ? `${horizontalInset}%` : "0";
+  } else if (horizontalAlign === "right") {
+    style.right = horizontalInset ? `${horizontalInset}%` : "0";
+  } else if (horizontalAlign === "center") {
+    style.left = "50%";
+    style.transform = "translateX(-50%)";
+  }
+
+  if (verticalAlign === "top") {
+    style.top = verticalInset ? `${verticalInset}%` : "0";
+  } else if (verticalAlign === "bottom") {
+    style.position = "fixed";
+    style.bottom = verticalInset ? `${verticalInset}%` : "0";
+  } else if (verticalAlign === "center") {
+    style.top = "50%";
+    style.transform = `${
+      horizontalAlign === "center" ? `${style.transform} ` : ""
+    }translateY(-50%)`;
+    style.display = "flex";
+    style.alignItems = "center";
+    style.justifyContent = "center";
+  }
+
+  if (height) {
+    style.height = `${height}vh`;
+  } else {
+    style.height = "100%";
+  }
+  return style;
+};
+
+export const mobileOverlay = mobileParameters => {
+  const { backdropOpacity, backdropColor } = mobileParameters;
+  const opacity = backdropOpacity || 0.5;
+  const color = backdropColor || "#FFFFFF";
+  const style = {
+    position: "fixed",
+    top: "0",
+    left: "0",
+    width: "100%",
+    height: "100%",
+    background: "transparent",
+    opacity,
+    backgroundColor: color
+  };
+  return style;
+};
+
+const REQUIRED_PARAMS = ["enabled", "parentElement", "insertionMethod"];
+
+const isValidWebParameters = webParameters => {
+  if (!webParameters) {
+    return false;
+  }
+
+  const ids = Object.keys(webParameters);
+
+  if (!includes(ids, ALLOY_MESSAGING_CONTAINER_ID)) {
+    return false;
+  }
+
+  if (!includes(ids, ALLOY_OVERLAY_CONTAINER_ID)) {
+    return false;
+  }
+
+  const valuesArray = values(webParameters);
+
+  for (let i = 0; i < valuesArray.length; i += 1) {
```

Comment:
use `forEach`

Reply:
we can't, because of the return statements within the loop